.github/workflows/mongodb.yml

```diff
@@ -0,0 +1,84 @@
+---
+name: "Tests: MongoDB"
+
+on:
+  pull_request:
+    branches: ~
+    paths:
+    - '.github/workflows/mongodb.yml'
+    - 'cratedb_toolkit/io/mongodb/**'
+    - 'pyproject.toml'
+  push:
+    branches: [ main ]
+    paths:
+    - '.github/workflows/mongodb.yml'
+    - 'cratedb_toolkit/io/mongodb/**'
+    - 'pyproject.toml'
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+  # Run job each night after CrateDB nightly has been published.
+  schedule:
+    - cron: '0 3 * * *'
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.8", "3.12"]
+        mongodb-version: ["2", "3", "4", "5", "6", "7"]
+
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: ${{ matrix.python-version }}
+      MONGODB_VERSION: ${{ matrix.mongodb-version }}
+      # Do not tear down Testcontainers
+      TC_KEEPALIVE: true
```

Comment:
It looks like `MONGODB_VERSION` may not be picked up by the test suite here. By default, it is `mongo:latest` [^1].

[^1]: https://github.com/testcontainers/testcontainers-python/blob/3add8cb09216/mongodb/testcontainers/mongodb/__init__.py#L51

Reply:
`MongoDbContainerWithKeepalive` has been improved correspondingly, and split into a seperate commit, see bf156ef52b1.